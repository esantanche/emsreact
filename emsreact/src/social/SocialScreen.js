import React, { Component } from 'react';

// Material UI components
import Grid from "@material-ui/core/Grid/Grid";

// Social networks logos
import GitHub_logo from '../images/GitHub_Logo.png';
import LinkedIn_logo from '../images/LinkedIn_logo.png';
import Medium_logo from '../images/Medium_logo.png';
import StackOverflow_logo from '../images/StackOverflow_logo.png';

// From components library
import SocialCard from "../components_library/cards/SocialCard";

/**
 * Showing four social network logos and linking them to the respective accounts.
 *
 * On small devices there will be two row with two logos each.
 *
 * On bigger devices there will be one row with four logos.
 */
class SocialScreen extends Component {

    render() {

        return (

            <React.Fragment>

                <Grid container spacing={16}>

                    <Grid item xs={6} sm={3} md={3} lg={3} >
                        <SocialCard logo={GitHub_logo} url_to_go="https://github.com/esantanche"/>
                    </Grid>

                    <Grid item xs={6} sm={3} md={3} lg={3} >
                        <SocialCard logo={Medium_logo} url_to_go="https://medium.com/@esantanche"/>
                    </Grid>

                    <Grid item xs={6} sm={3} md={3} lg={3} >
                        <SocialCard logo={LinkedIn_logo} url_to_go="https://www.linkedin.com/in/emanuelesantanche/"/>
                    </Grid>

                    <Grid item xs={6} sm={3} md={3} lg={3} >
                        <SocialCard logo={StackOverflow_logo} url_to_go="https://stackoverflow.com/users/6171615/emanuele-santanche"/>
                    </Grid>

                </Grid>

            </React.Fragment>

        )

    }

}

export default SocialScreen;
