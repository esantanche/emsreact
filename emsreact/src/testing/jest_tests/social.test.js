// Here we test components in the folder social

import React from "react";
import renderer from "react-test-renderer";

import SocialScreen from "../../social/SocialScreen";

describe('Testing SocialScreen', function() {

    const social_screen_to_be_tested = ( <SocialScreen /> );

    it('SocialScreen renders correctly', () => {

        const tree = renderer
            .create(social_screen_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});
