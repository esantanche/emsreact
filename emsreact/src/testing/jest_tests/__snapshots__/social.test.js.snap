// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SocialScreen SocialScreen renders correctly 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <a
        className="MuiButtonBase-root-127 SocialCard-cardButton-98"
        href="https://github.com/esantanche"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        target="_blank"
      >
        <div
          className="MuiCardMedia-root-130 SocialCard-image_responsive_size-97"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"GitHub_Logo.png\\")",
            }
          }
          title=""
        />
        <span
          className="MuiTouchRipple-root-132"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <a
        className="MuiButtonBase-root-127 SocialCard-cardButton-98"
        href="https://medium.com/@esantanche"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        target="_blank"
      >
        <div
          className="MuiCardMedia-root-130 SocialCard-image_responsive_size-97"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"Medium_logo.png\\")",
            }
          }
          title=""
        />
        <span
          className="MuiTouchRipple-root-132"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <a
        className="MuiButtonBase-root-127 SocialCard-cardButton-98"
        href="https://www.linkedin.com/in/emanuelesantanche/"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        target="_blank"
      >
        <div
          className="MuiCardMedia-root-130 SocialCard-image_responsive_size-97"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"LinkedIn_logo.png\\")",
            }
          }
          title=""
        />
        <span
          className="MuiTouchRipple-root-132"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-6-34 MuiGrid-grid-sm-3-45 MuiGrid-grid-md-3-59 MuiGrid-grid-lg-3-73"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <a
        className="MuiButtonBase-root-127 SocialCard-cardButton-98"
        href="https://stackoverflow.com/users/6171615/emanuele-santanche"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        target="_blank"
      >
        <div
          className="MuiCardMedia-root-130 SocialCard-image_responsive_size-97"
          src={undefined}
          style={
            Object {
              "backgroundImage": "url(\\"StackOverflow_logo.png\\")",
            }
          }
          title=""
        />
        <span
          className="MuiTouchRipple-root-132"
        />
      </a>
    </div>
  </div>
</div>
`;
